projects/serializers.py

```diff
@@ -86,7 +86,13 @@ class ProjectControlSerializer(serializers.ModelSerializer):
 
     class Meta:
         model = ProjectControl
-        fields = ("status", "project", "control", "catalog_data", "component_data", )
+        fields = ("status", "project", "control", "remarks", "catalog_data", "component_data", )
+
+    def validate(self, attrs):
+        remarks = attrs.get("remarks")
+        if attrs.get("status") == ProjectControl.Status.NA and (remarks is None or remarks == ""):
+            raise serializers.ValidationError("A justification is required for non-applicable controls.")
```

Comment:
Not sure where my original comment went, but I still don't understand why this was implemented in this way. I had suggested a more concise check:

```suggestion
        if attrs.get("status") == ProjectControl.Status.NA and not attrs.get("remarks"):
            raise serializers.ValidationError("A justification is required for non-applicable controls.")
```

Can you explain your reasoning?

Reply:
not sure, but I change it.